lib/change_stream.js

```diff
@@ -371,6 +371,8 @@ var processNewChange = function(self, err, change, callback) {
         .close()
```

Comment:
I think the problem here is that this method (`processNewChange`) handles recreation of the internal cursor on error, but then assumes that the user is using it not as a stream. As a result, we create the new cursor and then call `next` on it, and return it in a callback or a Promise. Right now you are returning that promise and not waiting for it to resolve, but registering the `data` listener in the `createChangeStreamCursor` _should_ be starting the initial find/getMore.